### PR TITLE
Transfer active processing refactoring

### PR DESCRIPTION
**Pull request by wxing1292**
* Rewrite some transfer active processing code
* Remove unused test utils

### CODE REVIEW

File: service/history/timerQueueActiveProcessor.go [hunk 1]
```diff
@@ -313,10 +313,8 @@ func (t *timerQueueActiveProcessorImpl) processUserTimerTimeout(
 	defer func() { release(retError) }()
 
 	mutableState, err := loadMutableStateForTimerTask(context, task, t.metricsClient, t.logger)
-	if err != nil {
+	if err != nil || mutableState == nil || !mutableState.IsWorkflowExecutionRunning() {
```
Comment (vancexu) [inline service/history/timerQueueActiveProcessor.go]: I feel it's more clear:
```
if err != nil {
return err
}
if workflowNotRunning() {
return nil
}
```

File: service/history/timerQueueProcessorBase.go [hunk 2]
```diff
@@ -403,22 +406,18 @@ func (t *timerQueueProcessorBase) processDeleteHistoryEvent(
 	defer func() { release(retError) }()
 
 	msBuilder, err := loadMutableStateForTimerTask(context, task, t.metricsClient, t.logger)
-	if err != nil {
+	if err != nil || msBuilder == nil || msBuilder.IsWorkflowExecutionRunning() {
```
Comment (vancexu) [inline service/history/timerQueueProcessorBase.go]: msBuilder -> mutableState?

File: service/history/transferQueueActiveProcessor.go [hunk 3]
```diff
@@ -417,42 +403,31 @@ func (t *transferQueueActiveProcessorImpl) processCloseExecution(
...
 	if err != nil {
 		return err
 	}
 	defer func() { release(retError) }()
 
-	var msBuilder mutableState
-	msBuilder, err = loadMutableStateForTransferTask(context, task, t.metricsClient, t.logger)
-	if err != nil {
+	mutableState, err := loadMutableStateForTransferTask(context, task, t.metricsClient, t.logger)
+	if err != nil || mutableState == nil || mutableState.IsWorkflowExecutionRunning() {
```
[the remark this answers is not in the document]
Comment (vancexu) [reply service/history/transferQueueActiveProcessor.go]: I have same feeling, as mentioned in my above comment.

File: service/history/timerQueueProcessorBase.go [hunk 4]
```diff
@@ -21,7 +21,7 @@
 package history
 
 import (
-	"context"
+	ctx "context"
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply service/history/timerQueueProcessorBase.go]: we use `context` as workflowExecutionContext, which usually collides with golang `context`

File: service/history/transferQueueActiveProcessor.go [hunk 5]
```diff
@@ -687,112 +569,93 @@ func (t *transferQueueActiveProcessorImpl) processCancelExecution(
 		ChildWorkflowOnly: common.BoolPtr(task.TargetChildWorkflowOnly),
 	}
 
-	op := func() error {
-		return t.historyClient.RequestCancelWorkflowExecution(nil, cancelRequest)
-	}
+	if err = t.requestCancelExternalExecutionWithRetry(cancelRequest); err != nil {
+		t.logger.Debug(fmt.Sprintf("Failed to cancel external workflow execution. Error: %v", err))
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply service/history/transferQueueActiveProcessor.go]: debug logging here is enough

File: service/history/transferQueueActiveProcessor.go [hunk 6]
```diff
@@ -1019,203 +878,187 @@ func (t *transferQueueActiveProcessorImpl) processUpsertWorkflowSearchAttributes
...
-		tag.WorkflowResetBaseRunID(resetPt.GetRunId()),
-		tag.WorkflowBinaryChecksum(resetPt.GetBinaryChecksum()),
-		tag.WorkflowEventID(resetPt.GetFirstDecisionCompletedId()))
 
 	resp, err := t.historyService.resetor.ResetWorkflowExecution(ctx.Background(), &workflow.ResetWorkflowExecutionRequest{
-		Domain:                common.StringPtr(domainEntry.GetInfo().Name),
-		WorkflowExecution:     &baseExecution,
-		Reason:                common.StringPtr(fmt.Sprintf("auto-reset reason:%v, binaryChecksum:%v ", reason, resetPt.GetBinaryChecksum())),
-		DecisionFinishEventId: common.Int64Ptr(resetPt.GetFirstDecisionCompletedId()),
+		Domain:            common.StringPtr(domainEntry.GetInfo().Name),
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply service/history/transferQueueActiveProcessor.go]: done for all signal / cancellation / start child workflow

**Pull request merged**